docs/bootstrap.md

```diff
@@ -0,0 +1,61 @@
+# Boostrap a new environment
+
+To rebuild the whole Foreman Infrastructure from scratch, these are the steps to get started.
+
+## Build a Puppetserver
+
+Install a minimal EL8 install. Then:
+
+```sh
+dnf install https://yum.puppet.com/puppet7-release-el-8.noarch.rpm
+dnf install puppetserver
+. /etc/profile.d/puppet-agent.sh
+puppetserver ca setup --ca-name 'Foreman Puppet CA' --certname $HOSTNAME --subject-alt-names puppet.theforeman.org
+puppet config set --section agent server puppet.theforeman.org
+puppet config set --section main dns_alt_names puppet.theforeman.org
+# To allow the foreman node a SAN
+sed -i '/allow-subject-alt-names/ s/false/true/' /etc/puppetlabs/puppetserver/conf.d/ca.conf
```

Comment:
(I guess we can enable that, sign the cert for `foreman.tfm.o` and disable it again?)

Reply:
You're right about that. The alternative I thought about was using Let's Encrypt for a proper URL. Tricky part there is access to it. Right now I have set up a firewall to only allow port 443 from the Puppetserver and my home network. I guess we could set up rules so that it does IP checks in Apache for everything except `/.well-known` or use DNS validation, but that's a lot trickier.

Reply:
I think I'd go with the easiest option, which to me seems to be "enable, sign, disable, (forget it 5 years)"?

Reply:
I'll opt to keep this in bootstrap for now, then when Puppet actually runs it should be disabled again.